python/src/aiconfig/editor/client/src/components/prompt/prompt_input/PromptInputRenderer.tsx

```diff
@@ -15,13 +18,34 @@ export default memo(function PromptInputRenderer({
   schema,
   onChangeInput,
 }: Props) {
-  return schema ? (
-    <PromptInputSchemaRenderer
-      input={input}
-      schema={schema}
-      onChangeInput={onChangeInput}
-    />
-  ) : (
-    <PromptInputConfigRenderer input={input} onChangeInput={onChangeInput} />
+  const [isRawJSON, setIsRawJSON] = useState(false);
+  return (
+    <>
+      <Flex justify="flex-end">
+        <Tooltip label="Toggle JSON editor" withArrow>
+          <ActionIcon onClick={() => setIsRawJSON((curr) => !curr)}>
+            {isRawJSON ? (
+              <IconBracesOff size="1rem" />
+            ) : (
+              <IconBraces size="1rem" />
+            )}
+          </ActionIcon>
+        </Tooltip>
+      </Flex>
+      {isRawJSON ? (
+        <PromptInputJSONRenderer input={input} onChangeInput={onChangeInput} />
+      ) : schema ? (
```

Comment:
If we're only allowing pure strings, why have this option at all in the input component?